### PR TITLE
Redirect category URLs to the Search Results page

### LINKED ISSUE #8094: Redirect category URLs to the Search Results page
This is a spin-off from #8092.

We currently support URLs like:
- https://addons.mozilla.org/en-US/firefox/extensions/alerts-updates/
- https://addons.mozilla.org/en-US/firefox/themes/fashion/

The Category pages that those URLs currently point to are going away, and we want to redirect any traffic to the URLs to the Search results page similar to what was done in https://github.com/mozilla/addons-frontend/issues/8079.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #8094

### CODE REVIEW

File: src/core/server/base.js [hunk 1]
```diff
@@ -258,6 +264,22 @@ function baseServer(
     });
   }
 
+  // Redirect category pages.
+  app.get(
```
Comment (bobsilverberg) [inline src/core/server/base.js]: Note that I didn't know how to write a test for this, but I did test it manually. I notice that we don't have any tests for the two `302` redirects in this file, so perhaps there isn't a straightforward way to test this?

Comment (willdurand) [reply src/core/server/base.js]: We should have test cases for that, at least for what has been added here.

**Comment by bobsilverberg** [comment]
> So, I thought about this patch and the new idea of moving nginx rules into this codebase. We should introduce a way to add and test those redirects, which is why I opened #8117.
> 
> I had this patch in mind when I wrote #8117, so this should work for this use case too. WDYT?

I took a quick look at #8117 and it looks good to me. I'll do a more detailed review and hopefully we can use that technique for this one as well.

**Comment by bobsilverberg** [comment]
@willdurand I adopted your approach for implementing and testing this. I had to change the test a bit because the `params` where not available when we called `buildNewURL` in order to generate a URL for the assertion, so I added them into the test definition structure. I played around with not having to define them twice (once in the `url` and once in `params`), by constructing the URL from the `params` in the body of the test, but it all ended up seeming a bit opaque.

I also kept your logic for testing for a string in the URL, even though I'm not currently using it, as it seems useful, but maybe we should remove it until such a time as we need it?

**Comment by bobsilverberg** [comment]
> > I also kept your logic for testing for a string in the URL, even though I'm not currently using it, as it seems useful, but maybe we should remove it until such a time as we need it?
> 
> I think you're right. We can remove it for now if we're not using it here.

Thanks @willdurand, I removed it and pushed a new commit. As discussed in IRC, might you be able to do another review of this to hopefully unblock me from the other patches I would like to finish and land?

**Pull request merged**